Automattic/wp-calypso
# Domains: Fix refunds for domain renewals

#### Changes proposed in this Pull Request

This PR removes a couple of checks that prevented domains from being canceled for the user:

* Domain renewals can be refunded from Calypso
* Domains of Atomic sites can be canceled
* Gets rid of the `Cancel Domain` button which simply disables auto-renew, and replaces it with the `Remove domain-name` button

Depends on D36311-code

#### Testing instructions

For verifying that Atomic domains can be canceled:

* Register a domain for an Atomic site
* Optionally set it as primary
* Go to purchases and you should be able to cancel the domain (there will be no refund for domains purchased with credits :))
* If you set the domain as primary, verify that the Atomic site is working fine (the primary domain should be *.wpcomstaging.com)

For verifying that domain renewals can be refunded:

* Using the store sandbox, purchase a domain
* Manually renew it at least once, ideally more than once
* Verify that you can refund the domain registration and the renewal(s)

Finally:

Purchase a domain with credits, and verify that the domain is not refundable in Calypso, but you are shown the option to remove the purchase (after which the subscriptions will be removed)
